ansible/post/files/approve_and_issue.sh

```diff
@@ -16,97 +16,97 @@ POWER_COUNT="${2}"
 # Var: ${self.triggers.approve}
 POWER_PREFIX="${3}"
 
-APPROVED_WORKERS=0
-ISSUED_WORKERS=0
+# Machine Prefix
+MACHINE_PREFIX="${POWER_PREFIX}-worker"
+
+# Check count for Power worker/s to be added is not zero
+if [ "0" -eq "${POWER_COUNT}" ]
+then
+  echo "Supplied Worker count is zero hence exiting."
+  exit 0
+fi
 
+# Setting values for variables
+APPROVED_WORKERS=0
 IDX=0
-while [ "$IDX" -lt "121" ]
+READY_COUNT=$(oc get nodes -l kubernetes.io/arch=ppc64le | grep "${MACHINE_PREFIX}" | grep -v NotReady | grep -c Ready)
+
+# Approce CSR and Check Ready status
+while [ "${READY_COUNT}" -ne "${POWER_COUNT}" ]
 do
+  
   export HTTPS_PROXY="http://${PROXY_SERVER}:3128"
 
-  echo "Try number: ${IDX}"
-  echo "List of Power Workers: "
+  echo "List of All Power Workers: "
   oc get nodes -l 'kubernetes.io/arch=ppc64le' -o json | jq -r '.items[] | .metadata.name'
   echo ""
 
+  echo "Approve and Issue - #${IDX}"
+  echo "List of Power Workers to be added with prefix '${POWER_PREFIX}': "
+  oc get nodes -l 'kubernetes.io/arch=ppc64le' --no-headers=true | grep "${MACHINE_PREFIX}"
+  echo ""
+
+  # Approve openshift-machine-config-operator:node-bootstrapper CSR/s
+  APPROVED_WORKERS=0
   JSON_BODY=$(oc get csr -o json | jq -r '.items[] | select (.spec.username == "system:serviceaccount:openshift-machine-config-operator:node-bootstrapper")' | jq -r '. | select(.status == {})')
   for CSR_REQUEST in $(echo ${JSON_BODY} | jq -r '. | "\(.metadata.name),\(.spec.request)"')
-  do 
+  do
     CSR_NAME=$(echo ${CSR_REQUEST} | sed 's|,| |'| awk '{print $1}')
     CSR_REQU=$(echo ${CSR_REQUEST} | sed 's|,| |'| awk '{print $2}')
     echo "CSR_NAME: ${CSR_NAME}"
     NODE_NAME=$(echo ${CSR_REQU} | base64 -d | openssl req -text | grep 'Subject:' | awk '{print $NF}')
-    echo "NODE_NAME: ${NODE_NAME}"
+    echo "Pending CSR found for NODE_NAME: ${NODE_NAME}"
 
-    if grep -q "system:node:${POWER_PREFIX}-worker-" <<< "$NODE_NAME"
+    if grep -q "system:node:${MACHINE_PREFIX}-" <<< "$NODE_NAME"
     then
-      echo ""
-      echo "${CSR_NAME}" | xargs -r oc adm certificate approve
+      oc adm certificate approve "${CSR_NAME}"
       APPROVED_WORKERS=$(($APPROVED_WORKERS + 1))
     fi
   done
 
+  # Approve system:node:${MACHINE_PREFIX} CSRs
   LOCAL_WORKER_SCAN=0
   while [ "$LOCAL_WORKER_SCAN" -lt "$POWER_COUNT" ]
```

Comment:
this can run in a perpetual loop... 
let's add 

L67
```suggestion
  while [ "$LOCAL_WORKER_SCAN" -lt "$POWER_COUNT" ]
  do
    SECOND_APPROVE=$(($SECOND_APPROVE + 1))
    if [ "${SECOND_APPROVE}" -eq "5" ]
    the
      break
    fi
    
```

Reply:
note, I think you'll want to add the above manually

Reply:
Regarding perpetual loop,
$LOCAL_WORKER_SCAN is getting incremented so once it's count is greater than $POWER_COUNT, while loop will be terminated. I have tried code in positive and negative scenarios.